src/Browscap/Coverage/Processor.php

```diff
@@ -0,0 +1,571 @@
+ * @license    MIT
+ */
+
+namespace Browscap\Coverage;
+
+use Seld\JsonLint\Lexer;
+
+/**
+ * Class Processor
+ *
+ * @category   Browscap
+ * @author     Jay Klehr <jay.klehr@gmail.com>
+ */
+class Processor
+{
+    /**
+     * @var string
+     */
+    private $resourceDir;
+
+    /**
+     * These are available in the JSON Parser part of the JSON Lint package we're using
+     * to Lexically analyze the JSON file, but they're a private property so can't re-use them directly
+     *
+     * @var array
+     */
+    private $symbols = [
+        'error' => 2,
+        'JSONString' => 3,
+        'STRING' => 4,
+        'JSONNumber' => 5,
+        'NUMBER' => 6,
+        'JSONNullLiteral' => 7,
+        'NULL' => 8,
+        'JSONBooleanLiteral' => 9,
+        'TRUE' => 10,
+        'FALSE' => 11,
+        'JSONText' => 12,
+        'JSONValue' => 13,
+        'EOF' => 14,
+        'JSONObject' => 15,
+        'JSONArray' => 16,
+        '{' => 17,
+        '}' => 18,
+        'JSONMemberList' => 19,
+        'JSONMember' => 20,
+        ':' => 21,
+        ',' => 22,
+        '[' => 23,
+        ']' => 24,
+        'JSONElementList' => 25,
+        '$accept' => 0,
+        '$end' => 1,
+    ];
+
+    /**
+     * @var array
+     */
+    private $symbolLookup = [];
+
+    /**
+     * @var array
+     */
+    private $coveredIds = [];
+
+    /**
+     * @var array
+     */
+    private $coverage = [];
+
+    /**
+     * @var int
+     */
+    private $funcCount = 0;
+
+    /**
+     * Create a new Coverage Processor for the specified directory
+     *
+     * @param string $resourceDir
+     */
+    public function __construct($resourceDir)
+    {
+        $this->resourceDir = $resourceDir;
+
+        $this->symbolLookup = array_flip($this->symbols);
+    }
+
+    /**
+     * Process the directory of JSON files using the collected pattern ids
+     *
+     * @param array $coveredIds
+     */
+    public function process(array $coveredIds)
+    {
+        $this->setCoveredPatternIds($coveredIds);
+
+        $iterator = new \RecursiveDirectoryIterator($this->resourceDir);
+
+        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
+            /** @var $file \SplFileInfo */
+            if (!$file->isFile() || $file->getExtension() !== 'json') {
+                continue;
+            }
+
+            $patternFileName = substr($file->getPathname(), strpos($file->getPathname(), 'resources/'));
+
+            if (!isset($this->coveredIds[$patternFileName])) {
+                $this->coveredIds[$patternFileName] = [];
+            }
+
+            $this->coverage[$patternFileName] = $this->processFile(
+                $patternFileName,
+                file_get_contents($file->getPathname()),
+                $this->coveredIds[$patternFileName]
+            );
+        }
+    }
+
+    /**
+     * Write the coverage data in JSON format to specified filename
+     *
+     * @param string $fileName
```

Comment:
So I added `@return void` to a lot of the changes in this PR for methods that don't return anything, and then ran `php-cs-fixer` on everything, which then removed them all due to this rule: `phpdoc_no_empty_return`, defined in the fixer config.

I'll remove that rule in php-cs-fixer config if there are no objections.